api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -131,6 +131,25 @@ default TableScan select(String... columns) {
    */
   Expression filter();
 
+  /**
+   * Create a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to {@param toSnapshotId} inclusive
+   * @param fromSnapshotId - the last snapshot id read by the user, exclusive
+   * @param toSnapshotId - read append data up to this snapshot id
+   * @return a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to {@param toSnapshotId} inclusive
+   */
+  TableScan appendsBetween(Long fromSnapshotId, Long toSnapshotId);
+
+  /**
+   * Create a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to the current snapshot id exclusive
+   * @param fromSnapshotId - the last snapshot id read by the user, exclusive
+   * @return a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to current snapshot inclusive
+   */
+  TableScan appendsAfter(Long fromSnapshotId);
```

Comment:
It looks like these snapshot IDs can now be `long` instead of being boxed because it wouldn't make sense to pass `null`.